### PR TITLE
Implemented admin functionalities with put and delete requests on books

### PR DESCRIPTION
**Pull request by adragicevicc**

### CODE REVIEW

File: src/components/core/Input/Input.tsx [hunk 1]
```diff
@@ -4,12 +4,13 @@ import classes from './Input.module.scss';
...
   onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  value?: string | number;
 }
 
-const Input = ({ name, onChange }: InputProps) => {
+const Input = ({ name, onChange, value }: InputProps) => {
   return (
     <div>
-      <input className={classes['c-input']} name={name} onChange={onChange}></input>
+      <input className={classes['c-input']} name={name} onChange={onChange} value={value}></input>
```
Comment (petarnqode) [inline src/components/core/Input/Input.tsx:13]: We should user /> on end when component don't have children

**Review by petarnqode — CHANGES_REQUESTED** [review]

File: src/components/BooksList/BooksList.tsx [hunk 2]
```diff
@@ -12,21 +12,53 @@ interface BooksListData {
   numOfCopies: number;
 }
 
-const BooksList = () => {
+type BooksListProps = {
+  filter?: string;
+};
```
Comment (vkotarlic) [inline src/components/BooksList/BooksList.tsx:17]: Here we should use Interface as we agreed on start

File: src/pages/Admin/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -4,7 +4,8 @@ import React from 'react';
 const Dashboard = () => {
   return (
     <div>
-      <BooksList />
+      Rented books:
+      <BooksList filter="rented-current" />
```
Comment (vkotarlic) [inline src/pages/Admin/Dashboard/Dashboard.tsx:8]: I wouldn't pass filter as prop here if that is possible

File: src/components/BookDialog/BookDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          <Input name="title" onChange={handleChange} value={book.title}></Input>
+        </div>
+        <div className={classes['c-book-dialog__input-container']}>
+          <label className={classes['c-book-dialog__input-label']}>Author</label>
+          <Input name="author" onChange={handleChange} value={book.author}></Input>
+        </div>
+        <div className={classes['c-book-dialog__input-container']}>
+          <label className={classes['c-book-dialog__input-label']}>Description</label>
+          <Input name="description" onChange={handleChange} value={book.description}></Input>
+        </div>
```
Comment (petarnqode) [inline src/components/BookDialog/BookDialog.tsx:37]: Can we refactor this code a bit, make one common component?

**Review by petarnqode — CHANGES_REQUESTED** [review]

File: src/components/BooksList/BooksList.tsx [hunk 5]
```diff
@@ -12,21 +12,53 @@ interface BooksListData {
   numOfCopies: number;
 }
 
-const BooksList = () => {
+type BooksListProps = {
+  filter?: string;
```
Comment (vkotarlic) [inline src/components/BooksList/BooksList.tsx:16]: Here we should type this as `'rented' | 'all'`

**Review by vkotarlic — CHANGES_REQUESTED** [review]

**Pull request merged**